src/Model/TransportModel/InterpolationScheme/UpwindScheme.f90

```diff
@@ -0,0 +1,60 @@
+module UpwindSchemeModule
+  use KindModule, only: DP, I4B
+  use ConstantsModule, only: DZERO
+  use IInterpolationSchemeModule, only: IInterpolationSchemeType, CoefficientsType
+  use BaseDisModule, only: DisBaseType
+  use TspFmiModule, only: TspFmiType
+
+  implicit none
+  private
+
+  public :: UpwindSchemeType
```

Comment:
Perhaps we should use the "upstream" terminology instead of "upwind" as that is more common for us?